DeDriftAndResample/DeDriftAndResamplePipeline.sh

```diff
@@ -27,1094 +27,616 @@
+fi
+log_Msg "After delimiter substitution, extractExtraRegNamesArr: ${extractExtraRegNamesArr[@]}"
 
-	if [ "${InRegName}" = "NONE" ]; then
-		InRegName=""
-	fi
-	
-	if [ -z "$MatlabRunMode" ]; then
-		MatlabRunMode="$G_DEFAULT_MATLAB_RUN_MODE"
-	fi
+if ((mrFIXExtractDoVolBool && ${#mrFIXExtractConcatNamesArr[@]} == 0))
+then
+	log_Err_Abort "--multirun-fix-extract-volume=TRUE requires --multirun-fix-concat-names"
+fi
 
-	# Log values retrieved from positional parameters
-	log_Msg "StudyFolder: ${StudyFolder}"
-	log_Msg "Subject: ${Subject}"
-	log_Msg "HighResMesh: ${HighResMesh}"
-	log_Msg "LowResMeshes: ${LowResMeshes}"
-	log_Msg "RegName: ${RegName}"
-	log_Msg "DeDriftRegFiles: ${DeDriftRegFiles}"
-	log_Msg "OutputRegName: ${OutputRegName}"
-	log_Msg "Maps: ${Maps}"
-	log_Msg "MyelinMaps: ${MyelinMaps}"
-	log_Msg "mrFIXNames: ${mrFIXNames}"
-	log_Msg "mrFIXConcatNames: ${mrFIXConcatNames}"
-	log_Msg "mrFIXExtractNames: ${mrFIXExtractNames}"
-	log_Msg "mrFIXExtractConcatNames: ${mrFIXExtractConcatNames}"
-	log_Msg "mrFIXExtractExtraRegNames: ${mrFIXExtractExtraRegNames}"
-	log_Msg "mrFIXExtractDoVol: ${mrFIXExtractDoVol}"
-	log_Msg "fixNames: ${fixNames}"
-	log_Msg "dontFixNames: ${dontFixNames}"
-	log_Msg "SmoothingFWHM: ${SmoothingFWHM}"
-	log_Msg "HighPass: ${HighPass}"
-	log_Msg "MotionRegression: ${MotionRegression}"
-	log_Msg "MyelinTargetFile: ${MyelinTargetFile}"
-	log_Msg "InRegName: ${InRegName}"
-	log_Msg "MatlabRunMode: ${MatlabRunMode}"
-
-	# Naming Conventions and other variables
-	local Caret7_Command="${CARET7DIR}/wb_command"
-	log_Msg "Caret7_Command: ${Caret7_Command}"
-
-	LowResMeshes=`echo ${LowResMeshes} | sed 's/@/ /g'`
-	log_Msg "After delimiter substitution, LowResMeshes: ${LowResMeshes}"
-
-	DeDriftRegFiles=`echo "$DeDriftRegFiles" | sed s/"@"/" "/g`
-	log_Msg "After delimiter substitution, DeDriftRegFiles: ${DeDriftRegFiles}"
-
-	Maps=`echo "$Maps" | sed s/"@"/" "/g`
-	log_Msg "After delimiter substitution, Maps: ${Maps}"
-
-    #these elses result in empty when given the empty string, make NONE do the same
-    if [[ "${MyelinMaps}" == "NONE" ]]
-    then
-        MyelinMaps=""
-    else
-    	MyelinMaps=`echo "$MyelinMaps" | sed s/"@"/" "/g`
+CorrectionSigma=$(echo "sqrt ( 200 )" | bc -l)
+log_Msg "CorrectionSigma: ${CorrectionSigma}"
+
+AtlasFolder="${StudyFolder}/${Subject}/MNINonLinear"
+log_Msg "AtlasFolder: ${AtlasFolder}"
+
+T1wFolder="${StudyFolder}/${Subject}/T1w"
+log_Msg "T1wFolder: ${T1wFolder}"
+
+NativeFolder="${AtlasFolder}/Native"
+log_Msg "NativeFolder: ${NativeFolder}"
+
+NativeT1wFolder="${T1wFolder}/Native"
+log_Msg "NativeT1wFolder: ${NativeT1wFolder}"
+
+ResultsFolder="${AtlasFolder}/Results"
+log_Msg "ResultsFolder: ${ResultsFolder}"
+
+#Naming Conventions
+DownSampleFolderNames=""
+DownSampleT1wFolderNames=""
+for LowResMesh in ${LowResMeshes} ; do
+	DownSampleFolderNames=`echo "${DownSampleFolderNames}${AtlasFolder}/fsaverage_LR${LowResMesh}k "`
```

Comment:
Could consider simplifying this to `DownSampleFolderNames+="${AtlasFolder}/fsaverage_LR${LowResMesh}k "`, or switching to arrays.

Reply:
Do not do `variable+="${variable}otherstuff "`, that will grow exponentially.